src/test/custom_nodes/node_do_nothing.cpp

```diff
@@ -0,0 +1,123 @@
+    if (inputsCount != 1) {
+        return 1;
+    }
+
+    if (strcmp(inputs[0].name, "input") != 0) {
+        return 2;
+    }
+
+    const struct CustomNodeTensor* input = &inputs[0];
+
+    *outputsCount = 1;
+    *outputs = (struct CustomNodeTensor*)malloc(sizeof(struct CustomNodeTensor) * (*outputsCount));
+    struct CustomNodeTensor* output = (&(*outputs))[0];
+
+    output->name = "output";
+    output->data = (uint8_t*)malloc(input->dataBytes * sizeof(uint8_t));
+    output->dataBytes = input->dataBytes;
+    memcpy((void*)output->data, (void*)input->data, input->dataBytes * sizeof(uint8_t));
+    output->dims = (uint64_t*)malloc(input->dimsCount * sizeof(uint64_t));
+    output->dimsCount = input->dimsCount;
+    memcpy((void*)output->dims, (void*)input->dims, input->dimsCount * sizeof(uint64_t));
+    output->precision = input->precision;
+    return 0;
+}
+
+// parse dims parameter from string representing list of ints, e.g. "dims" = "[3,5,10]"
+int parametrizeDimensions(std::vector<int>& dims, const struct CustomNodeParam* params, int paramsCount) {
+    for (int i = 0; i < paramsCount; i++) {
+        if (strcmp(params[i].key, "dims") == 0) {
+            std::string dimsString = params[i].value;
+            if (dimsString.front() != '[' || dimsString.back() != ']') {
+                std::cout << "dims param wrong format, should be [int, ...]" << std::endl;
+                return 1;
+            }
+            dims.clear();
+            // remove '[' and ']'
+            dimsString.erase(0, 1).pop_back();
+            size_t pos = 0;
+            while ((pos = dimsString.find(',')) != std::string::npos) {
+                dims.emplace_back(atoi(dimsString.substr(0, pos).c_str()));
+                dimsString.erase(0, pos + 1);
+            }
+            if (!dimsString.empty()) {
+                dims.emplace_back(atoi(dimsString.c_str()));
+            }
+        }
+    }
+    return 0;
+}
+
+int getInputsInfo(struct CustomNodeTensorInfo** info, int* infoCount, const struct CustomNodeParam* params, int paramsCount, void* customNodeLibraryInternalManager) {
+    std::vector<int> dims{3, 5, 10};
+    if (parametrizeDimensions(dims, params, paramsCount) != 0) {
+        return 1;
+    }
+
+    *infoCount = 1;
+    *info = (struct CustomNodeTensorInfo*)malloc(*infoCount * sizeof(struct CustomNodeTensorInfo));
+    (*info)->name = "input";
+    (*info)->dimsCount = dims.size();
+    (*info)->dims = (uint64_t*)malloc((*info)->dimsCount * sizeof(uint64_t));
+    for (int i = 0; i < dims.size(); i++) {
+        (*info)->dims[i] = dims.at(i);
+    }
+    (*info)->precision = FP32;
+    return 0;
+}
+
+int getOutputsInfo(struct CustomNodeTensorInfo** info, int* infoCount, const struct CustomNodeParam* params, int paramsCount, void* customNodeLibraryInternalManager) {
+    std::vector<int> dims{3, 5, 10};
```

Comment:
Why do we declare {3,5,10} vector if we fill it afterwards anyway?

Reply:
dropped dims declaring